### PR TITLE
Machinegod Cult [WIP][DO NOT MERGE]

### PR DESCRIPTION
**Pull request by VelardAmakar**
aye i'm putting stuff here to display progress and to have it all laid out when i finish
you can watch my progress here
get more info [here](https://docs.google.com/document/d/1Lkp-7MyzX3_BG1Gi7pDMJMZNE5SJTI0cCKlBIyrRXEI/edit#heading=h.5or1u1myu0h9) and [here](https://docs.google.com/spreadsheets/d/1yReHQohhT-a5QEdyfYARavV-UbuP3A3yadL11bNLFbo/edit#gid=127733079)
and when i have a forum thread up for this it'll be [here]()

i'm coding it, @RemieRichards is helping my dumb ass
@SkowronX and @FuryMcFlurry are doing sprites
s/o to the rest of #vgstation cause i wouldn't be able to do this without them too lmao
revenant

### CODE REVIEW

**Comment by dylanstrategie** [comment]
:^)

**Comment by VelardAmakar** [comment]
oh so now the old one wants to reopen

**Comment by clusterfack** [comment]
code/game/gamemodes/machinegod/machinegod_items.dm:518:error: red: undefined var
code/game/gamemodes/machinegod/machinegod_items.dm:541:error: c_animation: undefined var

**Comment by VelardAmakar** [comment]
what the actual fuck that combined on my end

File: code/game/gamemodes/machinegod/machinegod_items.dm [hunk 1]
```diff
@@ -0,0 +1,557 @@
...
+/obj/item/weapon/clockcheat/gateway
+	color = "#ffff00"
+
+/obj/item/weapon/clockcheat/gateway/attack_self(mob/living/user as mob)
+	if(!..()) return
+
+	var/list/L = list()
+	var/list/areaindex = list()
+
+	for(var/obj/machinery/clockobelisk/C in world)
```
Comment (ComicIronic) [inline code/game/gamemodes/machinegod/machinegod_items.dm]: Bad

File: code/game/gamemodes/machinegod/machinegod_structures.dm [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+		if(src in view(L)) //If you see it, get ready for brainfuck
+			if(prob(6)) L << "<span class='sinister'>Your sanity rapidly deteriorates as you look at the [src].</span>"
+			L.hallucination = max(L.hallucination, 10)
+
+		if(get_dist(src, L) <= 4)
+			if(L.getBrainLoss() >= 60)
+				if(prob(10))
+					bad_event(L)
+				if(iscultist(L) && prob(6))
```
Comment (ComicIronic) [inline code/game/gamemodes/machinegod/machinegod_structures.dm]: else, unless you want them to be able to be hit twice

File: code/modules/mob/living/simple_animal/clockcult.dm [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+	add_language(LANGUAGE_CLOCKCULT)
+	default_language = all_languages[LANGUAGE_CLOCKCULT]
+	for(var/spell in clockcult_spells)
+		src.add_spell(new spell, "clock_spell_ready")
+	updateicon()
+
+/mob/living/simple_animal/clockcult/Die()
+	..()
+	//new refuse item (src.loc)
+	for(var/mob/M in viewers(src, null))
```
Comment (ComicIronic) [inline code/modules/mob/living/simple_animal/clockcult.dm]: BUT WHY

visible_message

File: code/modules/mob/living/simple_animal/clockcult.dm [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+		if((M.client && !( M.blinded )))
+			M.show_message("<span class='warning'>[src] collapses in a shattered heap. </span>")
+	ghostize()
+	del src
+	return
+
+/mob/living/simple_animal/clockcult/examine(mob/user)
+	set src in oview()
```
Comment (ComicIronic) [inline code/modules/mob/living/simple_animal/clockcult.dm]: You don't need this, examine isn't a verb.

File: code/modules/mob/living/simple_animal/clockcult.dm [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+	if (src.health < src.maxHealth)
+		msg += "<span class='warning'>"
+		if (src.health >= src.maxHealth/2)
+			msg += "It appears somewhat exhausted.\n"
+		else
+			msg += "<B>It looks about ready to collapse!</B>\n"
+		msg += "</span>"
+	msg += "*---------*</span>"
+
+	usr << msg
```
Comment (ComicIronic) [inline code/modules/mob/living/simple_animal/clockcult.dm]: Why

File: code/modules/mob/living/simple_animal/clockcult.dm [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+
+
+/mob/living/simple_animal/clockcult/bound_revenant/Life()
+
+	//Health
+	if(active == MA_INACTIVE)
+		health += 2
+	else
+		health -= 1
+	if(!src in range(7,contractor.current))
```
Comment (ComicIronic) [inline code/modules/mob/living/simple_animal/clockcult.dm]: Just use get_dist here

**Comment by VelardAmakar** [comment]
all the stuff pertaining to mobs is preliminary copypaste shit that's almost entirely going to get overwrote

File: code/game/gamemodes/machinegod/machinegod_structures.dm (hunk 2 above)
Comment (VelardAmakar) [inline code/game/gamemodes/machinegod/machinegod_structures.dm]: that's the intent

File: code/game/gamemodes/machinegod/machinegod_items.dm (hunk 1 above)
Comment (VelardAmakar) [inline code/game/gamemodes/machinegod/machinegod_items.dm]: what should i do instead

Comment (PJB3005) [inline code/game/gamemodes/machinegod/machinegod_items.dm]: just make a global list list `var/global/list/clockobelisks = list()` and then add any obelisks to that, then add any obelisks to that.

make sure to remove them in Destroy() though

**Comment by DeityLink** [comment]
@Probe1 Man if not for that dakimakura that was perfect. (considering how many times @VelardAmakar came asking for my opinion when he came up with it)

File: code/game/gamemodes/machinegod/machinegod_items.dm (hunk 1 above)
Comment (DeityLink) [inline code/game/gamemodes/machinegod/machinegod_items.dm]: IN WORLD

**Comment by FudgePucker** [comment]
Will there be an option to choose what cult you perfer to be round start as?
Also I think it will be intersting how clockwork might be able to get protection from the station by helping out by flushing out the other cult, considering they greentext without having to summon a round ending laggy beast and most of their spells and devices can check for narsie followers.
'hey if you don't bash in our skulls with stun prods and let us rip out the brains of the other cult then we will not cause destruction upon the station.'

**Comment by VelardAmakar** [comment]
>Option to choose your cult

No. You'll get randomly assigned.

>Greentext without a summon

So does bloodcult, but it's in the same way that you might not be made to do it one round, and have to do it another.

>Devices can check for nar'sie followers

Only some, and not with the utmost accuracy. Plus, clockcult has the downside of being even harder to hide than bloodcult

**Comment by FudgePucker** [comment]
@VelardAmakar 
>No. You'll get randomly assigned.

Poop.
>bla

Will clockwork cults summon cause server crashing lag by converting everysingle tile to another tile?
>Only some

Sigil of Transgression's effect will be causing non cultists to stand still for 4 seconds and narsie dick suckers to fall over, right?
Put it down in the medbay hallway, when they stand still loyalty implant them, if they fall down, drag them to robotics, cut out their brain, pop it in a MMI, convert the MMI, and then borg/MoMMI/construct it. Being even harder to hide than bloodcult means you might perfer to offer your help to the station.

**Comment by VelardAmakar** [comment]
@FudgePucker 
>convert

I got a few ideas, whatever i do is gonna be similar but its honestly gonna be "similar but different"

>SoT

that's a very good point, i'm probably gonna make the sigil have a chance to damage all targets and for cultists instead of knocking them down they have a higher chance to receive damage

**Comment by SkowronX** [comment]
Tick tock

**Comment by clusterfack** [comment]
**when**

**Comment by SkowronX** [comment]
w h e n
h
e
n

**Comment by FuryMcFlurry** [comment]
When you code it

**Comment by VelardAmakar** [comment]
as much as i want to finish clockcult, this community is fucking garbage and i don't really want to be a part of it anymore
i've been thinking about it for a while and i decided on it a few days ago
the code for clockcult and all the documentation is there for anyone who wants it
if someone else wants to finish it, they can, but i'm not going to
thank you so much to everyone who has shown interest in clockcult and those who have contributed to it
i cannot thank you enough but i need to move on

**Comment by PJB3005** [comment]
F

**Comment by wwjnc** [comment]
F

**Comment by PJB3005** [comment]
Welp I always wanted to help with this so I guess I'll see about finishing it.

**Comment by SkowronX** [comment]
Velard may be out of commish, but I'll still do spritework.

**Pull request closed without merging**